plugins/modules/web_app_pool.ps1

```diff
@@ -0,0 +1,347 @@
+#!powershell
+
+# Copyright: (c) 2015, Henrik Wallström <henrik@wallstroms.nu>
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$ErrorActionPreference = 'Stop'
+
+$spec = @{
+    options = @{
+        name = @{
+            required = $true
+            type = "str"
+        }
+        state = @{
+            type = "str"
+            default = "present"
+            choices = @("absent", "present", "restarted", "started", "stopped")
+        }
+        attributes = @{
+            type = "dict"
+            # default empty dictionary to ensure it's always a dictionary
+            default = @{}
+        }
+    }
+    supports_check_mode = $true
+}
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$name = $module.Params.name
+$state = $module.Params.state
+$check_mode = $module.CheckMode
+$module.Result.info = @{
```

Comment:
I highly recommend we remove all these return values and don't return anything. This type of work should go into an `_info` module and not returning it here simplifies the code needed especially when it comes to check mode. The common practice for modules these days is to only return values that the user does not already know, for example a unique id of the resource they can use for references in subsequent tasks.